File: Class/C4/c4java/Student.java
import java.util.ArrayList;

public class Student {
    public String Name;
    public int Id;
    public ArrayList<Course> Courses;
    public ArrayList<Grade> Grades;
    public Student(String name, int id) {
        this.Name = name;
        this.Id = id;
        this.Courses = new ArrayList<Course>();
        this.Grades = new ArrayList<Grade>();
    }
    public String getInfo(){
        return this.Name + ' ' + this.Id;
    }
    public void addCourse(Course c){
        this.Courses.add(c);
    }
    public void addGrade(Grade g){
        this.Grades.add(g);
    }
    public void printCourses(){
        System.out.println("Courses for student " + this.Name + ": ");
        System.out.println();
        for(Course c : this.Courses)
        System.out.println(c.getInfo());
        System.out.println("***************************");
        System.out.println();
    }
    public void printGrades(){
        System.out.println("Grades for student " + this.Name + ": ");
        System.out.println();
        for(Grade g : this.Grades){
        System.out.print(g.Course.Name);
        System.out.print(": ");
        System.out.println(g.Value);
    }
    System.out.println("***************************");
    System.out.println();
    }
}

File: Exams/E1/cs/E1/Student.cs
using System;
class Student
{
    double[] Grades;
    int[] Weights;
    public Student(double[] G,int[] W)
    {
        Grades = new double[G.Length];
        Weights = new int[W.Length];
        for (int i = 0; i < W.Length; i++)
        {
            Grades[i] = G[i];
            Weights[i] = W[i];
        }
    }

    public double GetAvg()
    {
        double sum = 0;
        int w = 0;
        for (int i = 0; i < Grades.Length; i++)
        {
            sum+=Grades[i]*Weights[i];
            w+=Weights[i];
        }
        return sum/w;
    }
}